File: dotnet/src/Microsoft.Agents.Validation/Validators/ContentValidators/AudioContentValidator.cs
using Microsoft.Agents.Abstractions.Models;

namespace Microsoft.Agents.Validation.Validators.ContentValidators;

/// <summary>
/// Validator for AudioContent.
/// </summary>
public class AudioContentValidator : ContentValidatorBase<AudioContent>
{
    public override ValidationResult Validate(AudioContent content, ValidationContext? context = null)
    {
        var errors = new List<ValidationError>();

        // CNT-009: AudioContent duration must be positive
        var durationError = ValidatePositive(content.Duration, "Duration",
            ValidationErrorCode.CNT_009,
            "AudioContent duration must be a positive number");
        if (durationError != null)
            errors.Add(durationError);

        return new ValidationResult(errors);
    }
}

File: typescript/packages/agents-protocol-validation/src/validators/content/AudioContentValidator.ts
import type { AudioContent } from '@microsoft/agents-protocol-abstractions';
import type { ValidationContext } from '../../core/ValidationContext';
import { ContentValidatorBase } from '../base/ContentValidatorBase';
import { ValidationResult } from '../../core/ValidationResult';
import { ValidationError } from '../../core/ValidationError';
import { ValidationErrorCode } from '../../core/ValidationErrorCode';

/**
 * Validator for AudioContent.
 */
export class AudioContentValidator extends ContentValidatorBase<AudioContent> {
  public validate(content: AudioContent, context?: ValidationContext): ValidationResult {
    const errors: ValidationError[] = [];

    // Must have uri
    const uriError = this.validateNotEmpty(
      content.uri,
      'uri',
      ValidationErrorCode.CNT_006,
      'AudioContent must have uri'
    );
    if (uriError) {
      errors.push(uriError);
    }

    // Validate URI format
    if (content.uri) {
      const uriFormatError = this.validateUri(
        content.uri,
        'uri',
        ValidationErrorCode.CNT_006,
        'AudioContent uri must be a valid URI'
      );
      if (uriFormatError) {
        errors.push(uriFormatError);
      }
    }

    // mime-type should be audio/* (if present)
    if (content.mimeType && !content.mimeType.toLowerCase().startsWith('audio/')) {
      errors.push(
        new ValidationError(
          ValidationErrorCode.CNT_008,
          'AudioContent mime-type should start with "audio/"',
          'mimeType'
        )
      );
    }

    return new ValidationResult(errors);
  }
}
